### DISCUSSION TITLE
hooks for uniswap and others

### DISCUSSION THREAD

**Discussion by pociej** [wevm/wagmi#3106] [Ideas]
Im playing with uniswap contract in my app, and this create need for more specific hooks like for example 
`useSwapExactEthForToken` . Of course there is plenty of other usefull hooks that wraps uniswap standard methods but of course not only uniswap but other standards for example `useMintNFT` etc. I wonder would it be good idea to open PR here or such a hooks are to specific and its better to open separated repo?

**Comment by Feliza9** [comment]
I think there is no need to play with the Uniswap contract because it is a bad idea!

**Reply by EddyHoganu** [reply]
yeah people dont understand it

**Reply by pociej** [reply]
do not understand what? Uniswap exists so there is sometimes need to work with it. What are you guys talking about?

**Reply by Dimka524** [reply]
asdasdasdasd

**Reply by 98Kameron1902** [reply]
Can you explain why you can’t play with the Uniswap contract?

**Comment by NyleMc** [comment]
Hello! Uniswap stopped working for me. What should I do?

**Comment by r3ndyyy** [comment]
uniswap good

**Comment by Dimka524** [comment]
good
